sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_AzureEventSourceListenerTests.cs

```diff
@@ -0,0 +1,246 @@
+                }
+            }
+            finally
+            {
+                await producer.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task TraceListener()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:EventHubs_Sample10_TraceListener
+
+#if SNIPPET
+            var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+            var eventHubName = "<< NAME OF THE EVENT HUB >>";
+#else
+            var connectionString = EventHubsTestEnvironment.Instance.EventHubsConnectionString;
+            var eventHubName = scope.EventHubName;
+#endif
+
+            using AzureEventSourceListener traceListener = AzureEventSourceListener.CreateTraceLogger(EventLevel.LogAlways);
+
+            EventHubProducerClient producer = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                using var eventBatch = await producer.CreateBatchAsync();
+
+                var eventBody = new BinaryData("This is an event body");
+                var eventData = new EventData(eventBody);
+
+                if (!eventBatch.TryAdd(eventData))
+                {
+                    throw new Exception($"The event could not be added.");
+                }
+            }
+            finally
+            {
+                await producer.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task CustomListenerWithFilter()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:EventHubs_Sample10_CustomListenerWithFilter
+
+#if SNIPPET
+            var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+            var eventHubName = "<< NAME OF THE EVENT HUB >>";
+#else
+            var connectionString = EventHubsTestEnvironment.Instance.EventHubsConnectionString;
+            var eventHubName = scope.EventHubName;
+#endif
+
+            using AzureEventSourceListener customListener = new AzureEventSourceListener((args, message) =>
+            {
+                if (args.EventSource.Name.StartsWith("Azure-Identity") && args.Level == EventLevel.Verbose)
+                {
+                    Trace.WriteLine(message);
+                }
+                else if (args.EventSource.Name.StartsWith("Azure-Messaging-EventHubs"))
+                {
+                    switch (args.EventId)
+                    {
+                        case 3:
+                        case 4:
+                        case 5:
```

Comment:
```suggestion
                        case 3:   // Publish Start
                        case 4:   // Publish Complete
                        case 5:   // Publish Error
```

Reply:
For clarity, it'd be helpful to either leave a simple comment or declare some local `const` members to clarify the magic numbers.    Conceptually, this is spot-on!